src/core/cardano/walletConnect/identityWalletConnect.ts

```diff
@@ -45,15 +53,49 @@ class IdentityWalletConnect extends CardanoPeerConnect {
 
     this.sign = async (
       identifier: string,
-      payload: string
+      payload: string,
+      onSignError?: (error: Error) => void
     ): Promise<string> => {
-      if (this.signerCache.get(identifier) === undefined) {
-        this.signerCache.set(
+      let approved: boolean | undefined = undefined;
+      // Closure that updates approved variable
+      const approvalCallback = (approvalStatus: boolean) => {
+        approved = approvalStatus;
+      };
+      this.eventService.emit<PeerConnectSigningEvent>({
+        type: PeerConnectSigningEventTypes.PeerConnectSign,
+        payload: {
           identifier,
-          await Agent.agent.identifiers.getSigner(identifier)
-        );
+          payload,
+          approvalCallback,
+        },
+      });
+      let timeLeft = 3600000;
+      // Wait until approved is true or false
+      while (approved === undefined) {
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+        timeLeft -= 1000;
```

Comment:
2 things here:
1. Can we make 1000 a static variable in this class?
2. We should be measuring time rather than assuming it takes exactly 1000 every loop - just do something like:
```
const startTime = Date.now();
while (approved === undefined) {
  if (Date.now() > startTime + (IdentityWalletConnect.MAX_SIGN_TIME)) {
    ...
  }
}
```